src/main/java/com/ning/http/client/generators/InputStreamBodyGenerator.java

```diff
@@ -48,10 +48,17 @@ public InputStream getInputStream() {
 
     @Override
     public Body createBody() throws IOException {
-        if (inputStream.markSupported()) {
-          inputStream.mark(0);
+      if (inputStream.markSupported()) {
+        try {
           inputStream.reset();
+        } catch (IOException ioe) {
+          if (logger.isDebugEnabled()) {
+            logger.debug("Unable to reset the input stream: %s", ioe.getMessage());
```

Comment:
does `%s` work here? or have to use `{}`?

Reply:
Yes, you were right.